text/0000-dynamic-tag-names.md

```diff
@@ -0,0 +1,96 @@
+- Start Date: 2018-10-14
+- RFC PR: (leave this empty)
+- Ember Issue: (leave this empty)
+
+# Dynamic tag names in glimmer templates.
+
+## Summary
+
+With the transition from inner-html semantics to outer-html semantics in components, we lost one feature: Being
+able dynamically define the tag name of components.
+
+I think it was an useful feature and we should find a way to bring it back.
+
+## Motivation
+
+Although not something we use every day, there is a need for some components to have a dynamic tag name.
+
+This is most often used for certain low-level _presentational_ components.
+
+Take for instance a component named `<Panel>` that is used to encapsulate some presentation concerns.
+The template of that component could be like this:
+
+```hbs
+<div class="pt-10 pb-10 ps-20 box-shadow" ...attributes>
+  {{yield}}
+</div>
+```
+
+For accessibility and semantic reasons, sometimes a `<div>` may not be the best kind of tag.
+We might want the panel to be a `<section>` element, or an `<aside>` when it's content is somewhat unrelated
+with the rest of the page. Or maybe a `<legend>` if it's and the end of a form.
+
+## Detailed design
```

Comment:
I suggest a reordering of the paragraphs.

> We propose to add a new `element` helper that takes a tag name and generates a contextual component that, when invoked, renders the selected element.
> 
> Example:
> 
> ...
> 
> Unlike ids, classes or other attributes, the tag name of DOM element cannot be changed in runtime. ...
> 
> To help the user understand that changing the tag of an element in runtime is an expensive operation, the syntax is chosen such that...
> 
> Since the element name can be defined in runtime, in order to protect the user from XSS vulnerabilities, ...
>
> ...